pkg/trace/api/api.go

```diff
@@ -344,6 +344,7 @@ func (r *HTTPReceiver) loop() {
 			r.watchdog(now)
 		case now := <-t.C:
 			metrics.Gauge("datadog.trace_agent.heartbeat", 1, nil, 1)
+			metrics.Gauge("datadog.trace_agent.debug.trace_chan_len", float64(len(r.Out)), nil, 1)
```

Comment:
If we're going to end up tracking the length of multiple channels could be useful to name each channel with a tag while keeping the metric name constant for all channels, so something like `datadog.trace_agent.chan.length, chan:{chan_name}`, then we can do nice things like plot all agent channels grouped by name.

Reply:
I'd rather not mix unrelated channels. I feel a metric should make sense without narrowing down to a tag also, on it's own, as a total aggregation.